tests/integ/test_udf.py

```diff
@@ -1448,6 +1455,89 @@ def add_one_df_pandas_annotation(df: pandas.DataFrame) -> pandas.Series:
     )
 
 
+@pytest.mark.skipif(not is_pandas_and_numpy_available, reason="pandas is required")
+@pytest.mark.parametrize(
+    "_type, data, expected_type",
+    [
+        (IntegerType, [[4096]], "int16"),
```

Comment:
The `IntegerType` tests are consistently failing in all merge gates, claiming the expected type should be `object`, but locally it passes when running against `preprod3`. 
@sfc-gh-jdu Is this expected? 🤔

Reply:
hmm probably there is a parameter under the hood gating this behavior

Reply:
Linking relevant slack discussions for reference: [python-udf](https://snowflake.slack.com/archives/G01EKAN2FH9/p1663892558374559), [snowpark-python](https://snowflake.slack.com/archives/C02KH7E6SHX/p1663884904441309)